subsys/net/lib/bifravst_cloud/src/bifravst_cloud.c

```diff
@@ -0,0 +1,546 @@
+}
+
+static int topic_subscribe(void)
+{
+	const struct mqtt_subscription_list subscription_list = {
+		.list = (struct mqtt_topic *)&cc_rx_list,
+		.list_count = ARRAY_SIZE(cc_rx_list),
+		.message_id = SUBSCRIBE_ID
+	};
+
+	for (size_t i = 0; i < subscription_list.list_count; i++) {
+		LOG_DBG("Subscribing to: %s",
+		       log_strdup(subscription_list.list[i].topic.utf8));
+	}
+
+	return mqtt_subscribe(&client, &subscription_list);
+}
+
+static int publish_get_payload(struct mqtt_client *const c, size_t length)
+{
+	if (length > sizeof(payload_buf)) {
+		return -EMSGSIZE;
+	}
+
+	return mqtt_readall_publish_payload(c, payload_buf, length);
+}
+
+static void mqtt_evt_handler(struct mqtt_client *const c,
+			     const struct mqtt_evt *mqtt_evt)
+{
+	int err;
+	struct cloud_backend_config *config = bifravst_cloud_backend->config;
+	struct cloud_event cloud_evt = { 0 };
+
+#if defined(CONFIG_AWS_FOTA)
+	err = aws_fota_mqtt_evt_handler(c, mqtt_evt);
+	if (err > 0) {
+		/* Event handled by FOTA library so we can skip it */
+		return;
+	} else if (err < 0) {
+		LOG_ERR("aws_fota_mqtt_evt_handler, error: %d", err);
+		cloud_evt.type = CLOUD_EVT_ERROR;
+		cloud_notify_event(bifravst_cloud_backend, &cloud_evt,
+				   config->user_data);
+	}
+#endif
+
+	switch (mqtt_evt->type) {
+	case MQTT_EVT_CONNACK:
+		LOG_DBG("MQTT client connected!");
+
+		topic_subscribe();
+
+		cloud_evt.type = CLOUD_EVT_CONNECTED;
+		cloud_notify_event(bifravst_cloud_backend, &cloud_evt,
+				   config->user_data);
+		break;
+
+	case MQTT_EVT_DISCONNECT:
+		LOG_DBG("MQTT_EVT_DISCONNECT: result = %d", mqtt_evt->result);
+
+		cloud_evt.type = CLOUD_EVT_DISCONNECTED;
+		cloud_notify_event(bifravst_cloud_backend, &cloud_evt,
+				   config->user_data);
+		break;
+
+	case MQTT_EVT_PUBLISH: {
+		const struct mqtt_publish_param *p = &mqtt_evt->param.publish;
+
+		LOG_DBG("MQTT_EVT_PUBLISH: id = %d len = %d ",
+			p->message_id,
+			p->message.payload.len);
+
+		err = publish_get_payload(c, p->message.payload.len);
+		if (err) {
+			LOG_ERR("publish_get_payload, error: %d", err);
+			break;
+		}
+
+		if (p->message.topic.qos == MQTT_QOS_1_AT_LEAST_ONCE) {
+			const struct mqtt_puback_param ack = {
+				.message_id = p->message_id
+			};
+
+			mqtt_publish_qos1_ack(c, &ack);
+		}
+
+		cloud_evt.type = CLOUD_EVT_DATA_RECEIVED;
+		cloud_evt.data.msg.buf = payload_buf;
+		cloud_evt.data.msg.len = p->message.payload.len;
+
+		cloud_notify_event(bifravst_cloud_backend, &cloud_evt,
+				config->user_data);
+
+	} break;
+
+	case MQTT_EVT_PUBACK:
+		LOG_DBG("MQTT_EVT_PUBACK: id = %d result = %d",
+			mqtt_evt->param.puback.message_id,
+			mqtt_evt->result);
+		break;
+
+	case MQTT_EVT_SUBACK:
+		LOG_DBG("MQTT_EVT_SUBACK: id = %d result = %d",
+			mqtt_evt->param.suback.message_id,
+			mqtt_evt->result);
+		break;
+
+	default:
+		break;
+	}
+}
+
+#if defined(CONFIG_BIFRAVST_CLOUD_STATIC_IPV4)
+static int broker_init(void)
+{
+	int err;
+
+	struct sockaddr_in *broker =
```

Comment:
Could we change this variable name to `broker4`? Shadowing the global variable name makes it slightly confusing.